cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -386,22 +383,23 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
             throw new UnforceableProblem("Error getting pod " + nodeRef.podName(), e);
         }
 
-        if (!isPodStuck(pod)) {
+        restartContext.podStuck = isPodStuck(pod);
+        if (!restartContext.podStuck) {
             // We want to give pods chance to get ready before we try to connect to the or consider them for rolling.
             // This is important especially for pods which were just started. But only in case when they are not stuck.
             // If the pod is stuck, it suggests that it is running already for some time and it will not become ready.
             // Waiting for it would likely just waste time.
             LOGGER.debugCr(reconciliation, "Waiting for pod {} to become ready before checking its state", nodeRef.podName());
             try {
-                await(isReady(pod), operationTimeoutMs, TimeUnit.MILLISECONDS, e -> new RuntimeException(e));
+                await(isReady(pod), operationTimeoutMs, TimeUnit.MILLISECONDS, RuntimeException::new);
             } catch (Exception e) {
                 //Initialise the client for KafkaAgent if pod is not ready
                 if (kafkaAgentClient == null) {
                     this.kafkaAgentClient = initKafkaAgentClient();
                 }
                 BrokerState brokerState = kafkaAgentClient.getBrokerState(pod.getMetadata().getName());
                 if (brokerState.isBrokerInRecovery()) {
-                    throw new UnforceableProblem("Pod " + nodeRef.podName() + " is not ready because the broker is performing log recovery. There are  " + brokerState.remainingLogsToRecover() + " logs and " + brokerState.remainingSegmentsToRecover() + " segments left to recover.", e.getCause());
+                    throw new UnforceableProblem("Pod " + nodeRef.podName() + " is not ready because the broker is performing log recovery. There are " + brokerState.remainingLogsToRecover() + " logs and " + brokerState.remainingSegmentsToRecover() + " segments left to recover.", e.getCause());
```

Comment:
Can a controller do log recovery as well? Should we make the message more generic and call it node?